pkg/user/interactors/create_token.go

```diff
@@ -20,8 +20,11 @@ func NewCreateToken(logger logging.Logger, userProxy http.UserProxy) *CreateToke
 
 // Execute receives the user entity filled with e-mail and password properties and try
 // to create a token on the user proxy service. If it succeed, the token is returned.
-func (ct *CreateToken) Execute(user entities.User) (token string, err error) {
+func (ct *CreateToken) Execute(user entities.User, tokenType string) (token string, err error) {
 	token, err = ct.userProxy.CreateUserToken(user)
+	if err == nil && tokenType == "app" {
```

Comment:
If failed to create user token, the operation must be stopped, right? So, I think you can avoid validating if `err == nil` and move the `tokenType` checking to after the next if block.

```go
        if err != nil {
		ct.logger.Errorf("failed to create an user's token: %s", err.Error())
		return "", err
	}
        if tokenType == "app" {
                token, err = ct.userProxy.CreateAppToken(user.Email, token)
        }

        return token, nil


```

Reply:
You're right, @lcbm. The correct approach is to support this creation through another request that already comes with the user token. It's worth point this doesn't mean we need to implement two endpoints. What both of you think as a good design for that (cc. @lcbm, @esdrasjnr)?

Reply:
I'ts doable in the same endpoint but I still believe it should be two different operations. From what I understand, `type` parameter is already enough: 

1. check `type`
2. check paramters required for the appropriate `type`
3. return appropriate key or error 

We could also make the token duration configurable for api keys now, defaulting for never expiring if no `duration` params are passed.

Reply:
Hops [this](https://github.com/mainflux/mainflux/tree/master/authn) helps!

Reply:
Agreed, @lcbm. 😄

Reply:
Updated! Please, check if is there more improvements to be done :smile: